code/datums/status_effects/buffs.dm

```diff
@@ -627,6 +627,34 @@
 		H.physiology.burn_mod /=0.8
 		H.physiology.stamina_mod /= 0.8
 
+/datum/status_effect/breaching_and_cleaving
+	id = "breaching_and_cleaving"
+	duration = -1
+	status_type = STATUS_EFFECT_UNIQUE
+	alert_type = /atom/movable/screen/alert/status_effect/breaching_and_cleaving
+
+/atom/movable/screen/alert/status_effect/breaching_and_cleaving
+	name = "Breaching and Cleaving!"
+	desc = "<span class='danger'>Doors, people, machines... nothing will stand before your martial prowess!</span>"
+	icon_state = "breachcleaver"
+
+/datum/status_effect/breaching_and_cleaving/on_apply()
+	. = ..()
+	if(!ishuman(owner))
+		return
+	if(.)
+		var/mob/living/carbon/human/H = owner
+		var/datum/armor/cleavingArmorBoost = getArmor(0, 30, 30, 30, 0, 0, 50, 0, 0)
+		H.physiology.armor = H.physiology.armor.attachArmor(cleavingArmorBoost)
+		H.physiology.stamina_mod *= 0.8
+
+/datum/status_effect/breaching_and_cleaving/on_remove()
+	if(ishuman(owner))
+		var/mob/living/carbon/human/H = owner
+		var/datum/armor/cleavingArmorBoost = getArmor(0, 30, 30, 30, 0, 0, 50, 0, 0)
+		H.physiology.armor = H.physiology.armor.detachArmor(cleavingArmorBoost)
```

Comment:
```suggestion
		var/datum/armor/cleavingArmorBoost = getArmor(0, 30, 30, 30, 0, 0, 50, 0, 0)
		H.physiology.armor.attachArmor(cleavingArmorBoost)
		H.physiology.stamina_mod *= 0.8

/datum/status_effect/breaching_and_cleaving/on_remove()
	if(ishuman(owner))
		var/mob/living/carbon/human/H = owner
		var/datum/armor/cleavingArmorBoost = getArmor(0, 30, 30, 30, 0, 0, 50, 0, 0)
		H.physiology.armor.detachArmor(cleavingArmorBoost)
```
This should work the same?

Reply:
This does not. It was used earlier, but detachArmor and attachArmor don't actually mutate the value, and instead just return the new value. To attach and detach this, it needs to set the armor to the new value, thus the current implementation.